src/main/java/ddingdong/ddingdongBE/domain/notice/controller/NoticeApiController.java

```diff
@@ -1,31 +1,35 @@
 package ddingdong.ddingdongBE.domain.notice.controller;
 
+import ddingdong.ddingdongBE.domain.notice.api.NoticeApi;
+import ddingdong.ddingdongBE.domain.notice.controller.dto.request.GetAllNoticeByPageRequest;
 import ddingdong.ddingdongBE.domain.notice.controller.dto.response.DetailNoticeResponse;
-import ddingdong.ddingdongBE.domain.notice.controller.dto.response.NoticeResponse;
+import ddingdong.ddingdongBE.domain.notice.controller.dto.response.GetAllNoticeByPageResponse;
+import ddingdong.ddingdongBE.domain.notice.entity.Notice;
 import ddingdong.ddingdongBE.domain.notice.service.NoticeService;
-
 import java.util.List;
 import lombok.RequiredArgsConstructor;
-
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
-@RequestMapping("/server/notices")
 @RequiredArgsConstructor
-public class NoticeApiController {
+public class NoticeApiController implements NoticeApi {
 
     private final NoticeService noticeService;
 
     @GetMapping
-    public List<NoticeResponse> getAllNotices() {
-        return noticeService.getAllNotices();
+    public GetAllNoticeByPageResponse getAllNotices(GetAllNoticeByPageRequest request) {
+        List<Notice> notices = noticeService.getAllNotices(
+            request.getPage(),
+            request.getLimit()
+        );
+
+        return GetAllNoticeByPageResponse.from(notices);
     }
 
     @GetMapping("/{noticeId}")
-    public DetailNoticeResponse getNotice(@PathVariable Long noticeId) {
+    public DetailNoticeResponse getNoticeDetail(@PathVariable Long noticeId) {
```

Comment:
Response 이름 컨벤션에 맞게 변경도 같이 해주시면 좋을 것 같습니다
```suggestion
    public NoticeResponse getNoticeDetail(@PathVariable Long noticeId) {
```